src/bin/etl_twizzler/src/etl.rs

```diff
@@ -0,0 +1,289 @@
+use std::{
+    fs::File,
+    io::{self, BufRead, BufReader, Seek, SeekFrom},
+    path::PathBuf,
+};
+
+use serde::{Deserialize, Serialize};
+use tar::Header;
+#[cfg(target_os = "twizzler")]
+use twizzler_abi::{
+    object::{MAX_SIZE, NULLPAGE_SIZE},
+    syscall::{
+        sys_thread_sync, BackingType, LifetimeType, ObjectCreate, ObjectCreateFlags, ThreadSync,
+        ThreadSyncFlags, ThreadSyncReference, ThreadSyncWake,
+    },
+};
+#[cfg(target_os = "twizzler")]
+use twizzler_object::{ObjID, Object, ObjectInitFlags, Protections};
+
+// This type indicates what type of object you want to create, with the name inside
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone, Copy)]
+pub enum PackType {
+    // Create an object that is compatible with the twizzler std::fs interface, or the unix one
+    StdFile,
+    // Create raw twizzler object
+    TwzObj,
+    // Create a persistent vector object,
+    PVec,
+}
+
+pub struct Pack<T: std::io::Write> {
+    tarchive: tar::Builder<T>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+struct SpecialData {
+    kind: PackType,
+    offset: u64,
+}
+
+impl<W> Pack<W>
+where
+    W: std::io::Write,
+{
+    pub fn new(storage: W) -> Pack<W> {
+        let mut tarchive = tar::Builder::new(storage);
+        tarchive.mode(tar::HeaderMode::Deterministic);
+
+        Pack { tarchive }
+    }
+
+    pub fn file_add(
+        &mut self,
+        path: PathBuf,
+        pack_type: PackType,
+        offset: u64,
+    ) -> std::io::Result<()> {
+        let f = File::open(&path)?;
+
```

Comment:
A general overall comment.  Please reduce white space.  A blank line between every statement (or nearly every statement) is overkill.